Tests/AK/TestCharacterTypes.cpp

```diff
@@ -34,7 +36,31 @@ void compare_value_output_over(u32 range, auto& old_function, auto& new_function
     }
 }
 
-// NOTE: Avoid comparing over UNICODE in "TEST_CASE" due to test runtime becoming too long.
+void randomized_compare_bool_output_over(u32 range, auto& old_function, auto& new_function)
+{
+    // NOTE: randomized tests also run multiple times (100 by default). This means we'll try 10k random numbers times each time the test suite is run.
+    for (u32 n = 0; n < 100; ++n) {
+        bool result1 = false;
+        bool result2 = false;
+        GEN(i, Gen::unsigned_int(range - 1));
+        EXPECT_EQ(result1 = (old_function(i) > 0), result2 = (new_function(i) > 0));
+        if (result1 != result2)
+            dbgln("Function input value was {}.", i);
+    }
+}
+
+void randomized_compare_value_output_over(u32 range, auto& old_function, auto& new_function)
+{
+    // NOTE: randomized tests also run multiple times (100 by default). This means we'll try 10k random numbers times each time the test suite is run.
+    for (u32 n = 0; n < 100; ++n) {
+        i64 result1 = false;
+        i64 result2 = false;
+        GEN(i, Gen::unsigned_int(range - 1));
+        EXPECT_EQ(result1 = old_function(i), result2 = new_function(i));
+        if (result1 != result2)
+            dbgln("Function input value was {}.", i);
```

Comment:
We have a `FAIL` macro that one can pass a completely custom string into. This means something like the following should work:

```suggestion
        if ((result1 = old_function(i)) != (result2 = new_function(i))
            FAIL(String::formatted("New result {} does not match old result {} for input {}", result2, result1, i));
```